soroban-sdk/src/env.rs

```diff
@@ -750,20 +811,10 @@ impl Env {
     /// of the function invocation).
     ///
     /// The order of the returned vector is defined by the order of
-    /// `require_auth` calls. It is recommended though to do unordered
-    /// comparison in case if multiple entries are returned.
-    ///
-    /// 'Top-level call' here means that this is the first call of
-    /// `require_auth` for a given address in the call stack; it doesn't have
-    /// to coincide with the actual top-level contract invocation. For example,
-    /// if contract A doesn't use `require_auth` and then it calls contract B
-    /// that uses `require_auth`, then `verify_top_authorization` will return
-    /// `true` when verifying the contract B's `require_auth` call, but it will
-    /// return `false` if contract A makes a `require_auth` call.
-    ///
-    /// It is possible for a single address to be present multiple times in the
-    /// output, as long as there are multiple disjoint call trees for that
-    /// address.
+    /// [`require_auth`] calls. Repeated calls to [`require_auth`] in the same
```

Comment:
> Repeated calls to [`require_auth`] in the same tree of contract invocations will appear only once in the vector

This is not correct. First, trees are relative to the address, so for multiple addresses multiple entries will be present for the same contract invocation. Second, it's actually possible to call `reqiure_auth` for the same address multiple times in the same call - that's fine as long as the arguments are different.

Reply:
Fixed in d66eb844 and 53266a03. Thanks!